model/detection_options.go

```diff
@@ -0,0 +1,23 @@
+package model
+
+import "fmt"
+
+type GetAllOption func(query string, schemaPrefix string) string
+
+func WithEngine(engine EngineName) GetAllOption {
```

Comment:
Are the other unit tests already covering these returned strings?

Reply:
Not as directly as I'd like. I'll add unit tests to the options.